### PR TITLE
created shooter command

### PR DESCRIPTION
**Pull request by Weq511762338**
Created basic shooter subsystem

### CODE REVIEW

**Comment by Weq511762338** [comment]
I added the comments for each methods in the shooter subsystem

File: src/main/java/frc/robot/subsystems/Shooter.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+// suppress unneeded warnings about serialVersionUID in TreeMap
+@SuppressWarnings("serial")
+public class Shooter extends SubsystemBase {
+
+    //Shooter class constructor, initialize arrays for motors controllers, encoders, and SmartDashboard data
+    public Shooter() {
+
+    }
+
+    //periodically undate the values of motors for shooter to SmartDashboard
```
Comment (coachFitz0807) [inline src/main/java/frc/robot/subsystems/Shooter.java:35]: Typo: "undate" should be "update"

File: src/main/java/frc/robot/subsystems/Shooter.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    // Set the shooter and hood based on the distance
+    public void prepareShooter(double distance) {
+
+    }
+
+    //shoot the ball for high hub
+    public void shootHigh() {
+
+    }
+    //shoot the ball for low hub
```
Comment (coachFitz0807) [inline src/main/java/frc/robot/subsystems/Shooter.java:49]: We don't need shootLow(). dump() does the same thing.

**Review by coachFitz0807 — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/subsystems/Shooter.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+import com.revrobotics.CANSparkMaxLowLevel.MotorType;
+
+import edu.wpi.first.wpilibj.Servo;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+import frc.robot.Constants;
+
+// suppress unneeded warnings about serialVersionUID in TreeMap
+@SuppressWarnings("serial")
```
Comment (prensing) [inline src/main/java/frc/robot/subsystems/Shooter.java:27]: "suppresswarnings" should only be added when absolutely needed. Please remove

**Review by prensing — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/subsystems/Shooter.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+public class Shooter extends SubsystemBase {
+
+    //Shooter class constructor, initialize arrays for motors controllers, encoders, and SmartDashboard data
+    public Shooter() {
+
```
Comment (coachFitz0807) [inline src/main/java/frc/robot/subsystems/Shooter.java:32]: You need to add the motors and sensors. Per the spreadsheet, there will be two WPI_TalonFX motor controllers for the shooter, two CANSparkMax controllers for the hopper and let's assume for now, two limit switches to determine when there are two balls in the robot so the intake should not take in another ball.

**Review by coachFitz0807 — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/subsystems/Shooter.java [hunk 5]
```diff
@@ -12,38 +12,51 @@
...
 
-// suppress unneeded warnings about serialVersionUID in TreeMap
-@SuppressWarnings("serial")
+
 public class Shooter extends SubsystemBase {
 
+    //CANSparkMax for the hopper
+    CANSparkMax motor1, motor2;
+    //WPI_TalonSRX for the shooter
+    WPI_TalonSRX motor3, motor4;
```
Comment (coachFitz0807) [inline src/main/java/frc/robot/subsystems/Shooter.java:35]: This should be WPI_TalonFX

File: src/main/java/frc/robot/subsystems/Shooter.java [hunk 6]
```diff
@@ -12,38 +12,51 @@
...
 
 import frc.robot.Constants;
 
-// suppress unneeded warnings about serialVersionUID in TreeMap
-@SuppressWarnings("serial")
+
 public class Shooter extends SubsystemBase {
 
+    //CANSparkMax for the hopper
+    CANSparkMax motor1, motor2;
```
Comment (coachFitz0807) [inline src/main/java/frc/robot/subsystems/Shooter.java:33]: use "m_" prefix

File: src/main/java/frc/robot/subsystems/Shooter.java [hunk 7]
```diff
@@ -12,38 +12,51 @@
...
+    DigitalInput limitSwitch1, limitSwitch2;
+
+
     //Shooter class constructor, initialize arrays for motors controllers, encoders, and SmartDashboard data
     public Shooter() {
 
+        motor1 = new CANSparkMax(Constants.HOPPER_ONE_CAN_ID, MotorType.kBrushless);
+        motor2 = new CANSparkMax(Constants.HOPPER_TWO_CAN_ID, MotorType.kBrushless);
+
+        motor3 = new WPI_TalonSRX(Constants.SHOOTER_ONE_CAN_ID);
```
Comment (coachFitz0807) [inline src/main/java/frc/robot/subsystems/Shooter.java:46]: WPI_TalonFX

**Review by coachFitz0807 — CHANGES_REQUESTED** [review]

**Review by coachFitz0807 — APPROVED** [review]

**Review by prensing — APPROVED** [review]

**Pull request merged**